File: 231110/계속 중첩되는 사각형/continuously-overlapping-squares.py
n=int(input())

rec=[[0 for j in range(201)] for i in range(201)]

for i in range(1,n+1):
    x1,y1,x2,y2=map(int,input().split())
    for x in range(x1+100,x2+100):
            for y in range(y1+100,y2+100):
                if i%2!=0:
                    rec[x][y]=1
                else:
                    rec[x][y]=2


ans=0

for i in rec:
    for j in i:
        if j==2:
            ans+=1

print(ans)

File: 251207/계속 중첩되는 사각형/continuously-overlapping-squares.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const rects = input.slice(1).map((line)=>line.split(' ').map(Number));

const OFFSET = 100;
const MAX_R = 200;

const grid = Array.from({length:MAX_R+1}, ()=>Array(MAX_R+1).fill(0));

rects.forEach((rect,index)=>{
    let [x1,y1,x2,y2] = rect;
    x1+=OFFSET; y1+=OFFSET;
    x2+=OFFSET; y2+=OFFSET;
    for(let x=x1; x<x2; x++){
        for(let y=y1; y<y2; y++){
            if(index%2){ // blue
                grid[x][y]=2;
            }else{ // red
                grid[x][y]=1;
            }
        }
    }
})

let cnt=0;
for(let row of grid){
    cnt+= row.filter((e)=>e===2).length;
}

console.log(cnt)